pkg/locking/locking.go

```diff
@@ -118,7 +118,7 @@ func NewSession(cs *core.ControlSession, lmeta *LockingSPMeta, auth LockingSPAut
 	}
 
 	if err := auth.AuthenticateLockingSP(s, lmeta); err != nil {
-		return nil, fmt.Errorf("authentication failed: %v", err)
+		return nil, err
```

Comment:
@yogev-lb : please consider daisy-chaining using the original message with`%w`: some of the errors that `ThisSP_Authenticate` might pass back aren't self-evidently traced back to authentication and `AuthenticateLockingSP()` returns them verbatim, so knowing that the error happened on the authN path might come in handy.